src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -54,16 +55,21 @@ private static CipherAlgorithmType MapCipherAlgorithmType(long encryption)
                     return CipherAlgorithmType.Rc4;
                 case 8:
                     return CipherAlgorithmType.Rc2;
-                case 16:
-                    return CipherAlgorithmType.None;
                 case 32:
                     return CipherAlgorithmType.Null;
                 case 64:
                     return CipherAlgorithmType.Aes128;
                 case 128:
                     return CipherAlgorithmType.Aes256;
+                case 16:
+                case 256:
+                case 512:
+                case 1024:
+                case 2048:
+                case 4096:
+                case 8192:
+                    return (CipherAlgorithmType)encryption;              
```

Comment:
* Why do all of these values just inject onto the enum?
* Please create an enum to name the case labels in this switch; I have no idea what any of these values are supposed to be.
* Since this is a library with a separate contract (the `ref` directory project) the implementation library can add new enum values without their names being public.  So if Windows equivalents of these 7 values can be found, they can be correctly populated (which would match the Windows code's blind cast).
    * Failing that, making some local consts for what the Windows equivalents would be is better than having competing direct-cast implementations.